### PR TITLE
Chore: refactor credentials and cost

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Some more refactoring around credentials.
This is part of #394.

### CODE REVIEW

File: env0/data_credentials_test.go [hunk 1]
```diff
@@ -27,6 +27,18 @@ func TestCredentialsDataSource(t *testing.T) {
 			credentialsType: "AZURE_SERVICE_PRINCIPAL_FOR_DEPLOYMENT",
 			cloudType:       "azure",
 		},
+		{
```
Comment (TomerHeber) [inline env0/data_credentials_test.go:30]: Adding these cases allows me to remove "data_cost_credentials_test.go".

**Review by razbensimon — APPROVED** [review]
Nice one!

File: env0/data_credentials.go [hunk 2]
```diff
@@ -9,12 +9,12 @@ import (
...
-	"gcp":        {string(client.GcpServiceAccountCredentialsType)},
-	"azure":      {string(client.AzureServicePrincipalCredentialsType)},
-	"aws":        {string(client.AwsAssumedRoleCredentialsType), string(client.AwsAccessKeysCredentialsType)},
-	"gcp_cost":   {string(client.GoogleCostCredentialsType)},
-	"azure_cost": {string(client.AzureCostCredentialsType)},
-	"aws_cost":   {string(client.AwsCostCredentialsType)},
+	"gcp":         {string(client.GcpServiceAccountCredentialsType)},
+	"azure":       {string(client.AzureServicePrincipalCredentialsType)},
+	"aws":         {string(client.AwsAssumedRoleCredentialsType), string(client.AwsAccessKeysCredentialsType)},
+	"google_cost": {string(client.GoogleCostCredentialsType)},
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/data_credentials.go:15]: it's annoying... I agree...
The data source for some reason is called "env0_google_cost..."

I could add some "ifs" if preferred. (Not much better).

File: env0/credentials.go [hunk 3]
```diff
@@ -62,3 +65,31 @@ func getCredentials(id string, prefixList []string, meta interface{}) (client.Cr
...
+
+	id := d.Id()
+	err := apiClient.CloudCredentialsDelete(id)
+	if err != nil {
+		return diag.Errorf("could not delete credentials: %v", err)
+	}
+	return nil
+}
+
+func resourceCredentialsRead(cloudType string) schema.ReadContextFunc {
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/credentials.go:80]: yeah let me see what I can do.

File: env0/data_credentials.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (razbensimon) [reply env0/data_credentials.go:15]: Not a biggy here... 
@yaronya FYI maybe for an idea, future refactor.

File: env0/credentials.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/credentials.go:80]: @razbensimon , I'll push and fix it in the next PR.

**Pull request merged**